goodreads/templates/goodreads/csv_upload.html

```diff
@@ -2,10 +2,32 @@
 {% block content %}
 
 <script>
+function getCookie(name) {
+  let cookieValue = null;
+  if (document.cookie && document.cookie !== '') {
+      const cookies = document.cookie.split(';');
+      for (let i = 0; i < cookies.length; i++) {
+          const cookie = cookies[i].trim();
+          // Does this cookie string begin with the name we want?
+          if (cookie.substring(0, name.length + 1) === (name + '=')) {
+              cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+              break;
+          }
+        }
+  }
+    return cookieValue;
+  }
+
 function upload(){
+
+  const csrftoken = getCookie('csrftoken');
```

Comment:
I don't feel like I need the whole getCookie function but how else do I get the csrftoken ?

Reply:
this is fine - you can also get cookie parsing code from a library instead but this will do